crates/fluvio-spu/src/kv/consumer.rs

```diff
@@ -0,0 +1,295 @@
+use std::{
+    time::{Duration, SystemTime},
+    sync::Arc,
+    collections::{HashMap, hash_map::Entry},
+    ops::AddAssign,
+};
+
+use anyhow::Result;
+use async_rwlock::RwLock;
+use fluvio_kv_storage::KVStorage;
+use fluvio_protocol::{record::ReplicaKey, Encoder, Decoder};
+use fluvio_storage::FileReplica;
+use fluvio_types::PartitionId;
+use tracing::trace;
+
+use crate::replication::leader::{
+    LeaderKVStorage, FollowerNotifier, LeaderReplicaState, LeaderReplicaLog,
+};
+
+pub(crate) type TimestampSecs = u64;
+
+const DEFAULT_FLUSH_THRESHOLD: usize = 100;
+
+#[derive(Debug, Default)]
+pub(crate) struct SharedConsumerStorages(Arc<RwLock<HashMap<ReplicaKey, SharableConsumerStorage>>>);
+
+#[derive(Debug, Clone)]
+pub(crate) struct SharableConsumerStorage(Arc<RwLock<ConsumerStorage>>);
+
+#[derive(Debug, Hash, Clone, Default, PartialEq, Eq, PartialOrd, Ord, Encoder, Decoder)]
+pub(crate) struct Key {
+    replica_id: ReplicaKey,
+    consumer_id: String,
+}
+
+#[derive(Debug, Default, Clone, PartialEq, Eq, PartialOrd, Ord, Encoder, Decoder)]
+pub(crate) struct Consumer {
```

Comment:
Can you rename this to ConsumerOffset or ConsumerSession.  Not sure what this tracks